bundles/config/org.eclipse.smarthome.config.dispatch/src/main/java/org/eclipse/smarthome/config/dispatch/internal/ConfigDispatcher.java

```diff
@@ -335,16 +373,49 @@ private void processConfigFile(File configFile) throws IOException, FileNotFound
                 logger.warn("The file {} subsequently defines the exclusive PID '{}'.", configFile.getAbsolutePath(),
                         exclusivePID);
             }
+
             pid = exclusivePID;
+
+            if (exclusivePID.contains(SERVICE_CONTEXT_MARKER)) {
+                // split pid and context
+                pid = exclusivePID.split(SERVICE_CONTEXT_MARKER)[0];
+                context = exclusivePID.split(SERVICE_CONTEXT_MARKER)[1];
+            }
+
             lines = lines.subList(1, lines.size());
-            exclusivePIDMap.setProcessedPID(pid, configFile.getAbsolutePath());
+            exclusivePIDMap.setProcessedPID(exclusivePID, configFile.getAbsolutePath());
+
         } else if (exclusivePIDMap.contains(pid)) {
             // the pid was once from an exclusive file but there is either a second non-exclusive-file with config
             // entries or the `pid:` marker was removed.
             exclusivePIDMap.removeExclusivePID(pid);
         }
 
-        Configuration configuration = configAdmin.getConfiguration(pid, null);
+        Configuration configuration = null;
+        // if we have a context we need to create a service factory
+        if (context != null) {
+            try {
+                // try to find configuration using our context property
+                configuration = getConfigurationWithContext(exclusivePID);
+            } catch (InvalidSyntaxException e) {
+                logger.error("Failed to lookup config for file '{}' for PID '{}' with context '{}'",
+                        configFile.getName(), pid, context);
+            }
+
+            if (configuration == null) {
+                logger.debug("Creating factory configuration for PID '{}' with context '{}'", pid, context);
+
+                configuration = configAdmin.createFactoryConfiguration(pid, null);
+                Dictionary p = new Properties();
+                p.put(SERVICE_CONTEXT, context);
+                configuration.update(p);

```

Comment:
This only works because the config is not overridden by the check in line 422: When you correct the check there to also take the context into account the `SERVICE_CONTEXT` property set here will be gone. May be the `SERVICE_CONTEXT` should be set around the check in line 422 for exclusive PIDs with context, right after creating a blank dictionary for this configuration.

Reply:
Since we dont talk about overwriting an existing exclusive config any more I like to stress this _issue_ again: This works only because the check in line 422 checks for the `pid` which is stripped from the context in this case.
I would like it better to have the `new Properties()` and `p.put(SERVICE_CONTEXT, context)` somewhere around the check if we are dealing with an exclusive PID.

Reply:
I have put the creating of the new properties dictionary near the place where the empty properties are created for the non-context cases.